File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        int[] nums = Arrays.stream(br.readLine().split(" "))
                .mapToInt(Integer::parseInt)
                .toArray();

        int left = 0;
        int right = 0;
        int count = 0;
        int current = 0;

        while (right < n) {
            if (current < m) {
                current += nums[right];
                right++;
            } else if (current == m) {
                count++;
                current -= nums[left];
                left++;
            } else {
                current -= nums[left];
                left++;
            }
        }

        while (left < n) {
            if (current == m) {
                count++;
            }
            current -= nums[left];
            left++;
        }

        System.out.println(count);
    }
}

File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.kt
/**
 * p2003 수들의 합2
 */
fun main() {
    val (n, m) = readln().split(" ").map { it.toInt() }
    val nums = readln().split(" ").map { it.toInt() }.toIntArray()

    var left = 0
    var right = 0
    var count = 0
    var currentSum = 0

    while (right < n) {
        when {
            currentSum < m -> {
                currentSum += nums[right]
                right++
            }
            currentSum == m -> {
                count++
                currentSum -= nums[left]
                left++
            }
            else -> {
                currentSum -= nums[left]
                left++
            }
        }
    }

    while (left < n) {
        if (currentSum == m) {
            count++
        }
        currentSum -= nums[left]
        left++
    }

    println(count)
}